main.nf

```diff
@@ -538,13 +547,20 @@ process phantompeakqualtools {
                 saveAs: {filename -> filename.indexOf(".out") > 0 ? "logs/$filename" : "$filename"}
 
     input:
-    file bam from bam_dedup_spp
+    file bam_dedup from bam_dedup_spp
+    file bam_total from bam_total_spp
 
     output:
     file '*.pdf' into spp_results
     file '*.spp.out' into spp_out, spp_out_mqc
 
     script:
+    if(!params.skipDupRemoval){
```

Comment:
Instead of putting this logic in all of the downstream processes (and staging loads of files), it's better to manipulate the channels. For an example, see https://github.com/nf-core/methylseq/blob/93600734bc7a1c1a12a3848b4429425ee3aee79c/main.nf#L314-L317

Basically, wrap the entire optional process in an `if` block and have the `else` create the expected output channel name directly from the input channel name. Then, all downstream processes can just use the expected output channel as normal.